tools/kagami/src/crypto.rs

```diff
@@ -1,14 +1,135 @@
-use clap::{builder::PossibleValue, ArgGroup, ValueEnum};
-use color_eyre::eyre::WrapErr as _;
+use std::{
+    borrow::{Borrow, Cow},
+    fs,
+    path::{Path, PathBuf},
+};
+
+use clap::{builder::PossibleValue, ArgGroup, Subcommand, ValueEnum};
+use color_eyre::{
+    eyre::{eyre, WrapErr as _},
+    Result,
+};
 use iroha_crypto::{Algorithm, KeyPair, PrivateKey};
+use iroha_data_model::ChainId;
+use iroha_genesis::{GenesisNetwork, RawGenesisBlock};
+use parity_scale_codec::Encode;
 
 use super::*;
 
+#[derive(Subcommand, Debug, Clone)]
+pub enum CryptoMode {
+    GenesisSigning(GenesisSigningArgs),
+    KeyPairGeneration(KeyPairArgs),
+}
+
+/// Use `Kagami` to sign genesis block.
+#[derive(ClapArgs, Clone, Debug)]
+#[command(group = ArgGroup::new("private_key").required(true))]
+#[command(group = ArgGroup::new("public_key").required(true))]
+#[command(group = ArgGroup::new("format").required(false))]
+pub struct GenesisSigningArgs {
+    /// The algorithm of the provided keypair
+    #[clap(default_value_t, long, short)]
+    algorithm: AlgorithmArg,
+    /// Private key (in string format) to sign genesis block
+    #[clap(long, group = "private_key")]
+    private_key_string: Option<String>,
+    /// Path to private key to to sign genesis block
+    #[clap(long, group = "private_key")]
+    private_key_path: Option<PathBuf>,
```

Comment:
```suggestion
    private_key_file: Option<PathBuf>,
```

In the configuration, we agreed to use `_file` suffix for file paths, and `_dir` for dir paths. Let's follow this convention everywhere else as well.